### ISSUE TITLE
Show how to delete and move files

### ISSUE THREAD

**Issue by KathleenLabrie** [swcarpentry/git-novice#111]
This tutorial almost perfect for my target audience.  The only thing missing, I think, is to show how to properly delete and move files around.  Should be fairly straightforward to add something like that to the "change" section.

**Comment by r-gaia-cs** [comment]
@KathleenLabrie Thanks for the report.

> The only thing missing, I think, is to show how to properly delete and move files around.

Git is very smart and identify deleted and moved files:

~~~
$ git init
Initialized empty Git repository in /tmp/sample/.git/
$ touch foo.txt
$ git add foo.txt 
$ git commit -m 'Create foo.txt'
[master (root-commit) e6d02a3] Create foo.txt
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 foo.txt
$ mv foo.txt bar.txt
$ git status
On branch master
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    foo.txt

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        bar.txt

no changes added to commit (use "git add" and/or "git commit -a")
$ git add foo.txt bar.txt 
$ g status
On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        renamed:    foo.txt -> bar.txt
~~~

> Should be fairly straightforward to add something like that to the "change" section.

Can you send a pull request with the challenge? If yes and you need help, let us know.

**Comment by gvwilson** [comment]
We should introduce `git mv` and `git rm`, and also say that Git can 
figure out moves and deletes.

**Comment by iglpdc** [comment]
I'm -1 to add this to the novice lesson, although I agree it's valuable. I think that moving and deleting files is not something novices will face the first few times they use a repo, and the lesson it's pretty hard to teach in 3 hours already.

If we want to include move and delete, we will need to explain `git rm` and `git mv` (which involves at least 3 commits, see the amount of code in @r-gaia-cs 's example above), and why/how Git tracks the file when we use just `mv` (or maybe a bit on how Git keeps track of changes). I think this will add another 5-10 minutes to the lesson and will probably require another topic.

This could be good for the intermediate lesson or extras in this topic, depending on how deep we want to go on explaining how Git does the things under hood.

**Comment by KathleenLabrie** [comment]
My point is that even novice users will at one point delete or rename a file.  Yet, all they know about is "git add" and that doesn't work to stage a file for removal.  One needs to use "git rm".  @r-gaia-cs, a simple "git add foo.txt bar.txt" doesn't work for me for the rename.  I need to use "git add --all foo.txt bar.txt".  Any idea why I need "--all" and you don't?

Also, doing the file deletion or renaming with "git rm" and "git mv" is obviously a good idea, however most novice, and not so novice users, are more likely to use the Unix command directly.  I think we need to show them how to stage and commit those changes to the repository.

Anyway, to answer @r-gaia-cs, I unfortunately do not have much time for this now or in the coming weeks.  The "real" day job is getting in the way.  I just thought that for now I'd bring the issue to people's attention and see how other people felt about this.   I have no experience whatsoever teaching git to anyone, so comments like @iglpdc 's are very important to consider.

**Comment by iglpdc** [comment]
I guess @r-gaia-cs is using a newer version of Git where `--all` is no longer needed.

In any case, I still think this is a good addition, but as an extra and in a separate topic. It's true that `git status` tells you to use `git rm` to update the index, so that's my only concern. 

We could make an "Undoing things" topic (and decide later if it's an extra or another thing), consisting of:
- undoing "files": `git rm, git mv` *vs* Unix `mv, rm`
- undoing changes: basically developing the diagram in #88 

Thoughts?

**Comment by r-gaia-cs** [comment]
> Any idea why I need --all and you don't?

As @iglpdc suggest, probably Git version:

~~~
$ git --version
git version 2.3.7
~~~

Normally I'm using the last version of the tool available. =)

> I unfortunately do not have much time for this now

No problem at all. Thanks a lot for the feedback.

**Comment by KathleenLabrie** [comment]
@iglpdc That's a great diagram, but way out of reach of novices.  I was thinking on much more basic terms.  Something like this for deleting files:

    $ echo "This is another name for fake planet X" > nibiru.txt
    $ git add nibiru.txt
    $ git commit -m 'adding info on nibiru'
    $ git status
    On branch master
    nothing to commit, working directory clean
    $ rm nibiru.txt
    $ git status
    On branch master
    Changes not staged for commit:
       (use "git add/rm <file>..." to update what will be committed)
       (use "git checkout -- <file>..." to discard changes in working directory)

	    deleted:    nibiru.txt

    no changes added to commit (use "git add" and/or "git commit -a")
    $ git rm nibiru.txt
    $ git status
    On branch master
    Changes to be committed:
      (use "git reset HEAD <file>..." to unstage)

	    deleted:    nibiru.txt

    $ git commit -m 'removing info on Nibiru, it is not a real planet!'

Then something similar for a file rename:

    $ echo "Superman's home planet" > krypton.txt
    $ git add krypton.txt
    $ git commit -m 'Adding planet Krypton'
    $ mv krypton.txt earth.txt
    $ git status
    On branch master
    Changes not staged for commit:
      (use "git add/rm <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)

    	    deleted:    krypton.txt

    Untracked files:
      (use "git add <file>..." to include in what will be committed)

	    earth.txt

    no changes added to commit (use "git add" and/or "git commit -a")
    $ git add krypton.txt earth.txt
    $ git status
    On branch master
    Changes to be committed:
      (use "git reset HEAD <file>..." to unstage)

	    renamed:    krypton.txt -> earth.txt

    $ git commit -m "Superman's home is now Earth"

**Comment by KathleenLabrie** [comment]
Yeah!  my git is insanely old it seems.  :) 

    $ git --version
    git version 1.8.5.2 (Apple Git-48)

**Comment by iglpdc** [comment]
That's a great example and fits pretty well the history of the lesson. You should definitely make a PR with this (when you have time, of course.)

As I said, the only worry is time. The lesson takes already a good 3 hours, and I feel this is less important that what we have now.  

We still have to decide how to decide how to treat this extra materials, things that could be taught to novices but don't fit in for time reasons, but this could be a good addition to such materials.

**Comment by KathleenLabrie** [comment]
I takes 3 hours to teach this.   Wow!  I would have never thought.  Good to know.  Not a brown bag lunch sort of thing then.  

Do you have suggestions on where that example I propose could live?   I'm new to SWCarpentry and don't know the styles and preferences yet.

**Comment by iglpdc** [comment]
> I takes 3 hours to teach this. Wow! I would have never thought. Good to know. Not a brown bag lunch sort of thing then.

Yes, we do it as a hands-on tutorial, and takes about that. If you look at the beginning of each Markdown file, there's an estimate for the duration of the topic.

> Do you have suggestions on where that example I propose could live? I'm new to SWCarpentry and don't know the styles and preferences yet.

This is something that we are debating a bit. Right now, we're collecting this bits as extras (things that cannot fit in the lesson, but still are at novice level) in the `discussion.md` file. So, if you want to make the contribution now, the best would be add two subsections (say, "Deleting files" and "Moving files") at the end of the file. 

In the future (next couple of weeks), we will probably reorganize these materials somehow: maybe leaving the `discussion.md` file for shorter entries and having links to specific short files for entries a big longer like yours.

In any case, if you make a PR, just include the changes to the markdown files, we generate the html separately.

**Comment by KathleenLabrie** [comment]
Got it!  Given that you are planning a re-organization soon, I'll try to add my example to `discussion.md` asap.  I've already written the examples in a previous comment anyway.  I'll add a explanation text around that.

**Comment by gvwilson** [comment]
+1, and thank you.

**Comment by iglpdc** [comment]
Closed by #146.
